js/main.js

```diff
@@ -1,2 +1,96 @@
 /* global data */
 /* exported data */
+// the form
+var $entryForm = document.getElementById('entry-form');
+var $entryImage = document.querySelector('.placeholder');
+// the form inputs
+var $inputImageURL = document.querySelector('#photo-url');
+var $inputTitle = document.querySelector('#title');
+var $inputNotes = document.querySelector('#notes');
+
+// Event listener to update img src with input values
+$inputImageURL.addEventListener('change', updateImage);
+
+// function to insert the value of the input URL into the placeholder image src)
+function updateImage(event) {
+  // update the class of the placeholder image to .entry-image. Remove placeholder class & alt text
+  $entryImage.classList.add('entry-image');
+  $entryImage.alt = '';
+  $entryImage.src = event.target.value;
+  $inputImageURL = event.target.value;
+  // update the src attribute of the image to the $inputImageURL value
+  return $entryImage;
+}
+// event listener for submit event on journal entry form
+$entryForm.addEventListener('submit', saveEntry);
+function saveEntry(event) {
+  event.preventDefault();
+  $inputNotes = $entryForm.elements.notes.value;
+  $inputTitle = $entryForm.elements.title.value;
+  data.entries.push({
+    title: $inputTitle,
+    photoURL: $inputImageURL,
+    notes: $inputNotes
+  });
+  data.nextEntryId++;
+  $entryImage.src = 'images/placeholder-image-square.jpg';
+  $entryForm.reset();
+}
+
+// function to build out the basic node structure for the entries
+
+// <li>
+// <div class="entries row">
+//    <div class="column-half">
+//      <img src="$entryImage.src" alt=""></div>
+//    <div class="column-half">
+//      <h2>$$inputTitle</h2>
+//     <p>$$inputNotes</p>
+//    </div>
+//  </div>
+// </li> */
+
+// Define the parent element
+var $entriesList = document.getElementsByClassName('entries-list');
+// Create the first child element and append it to parent element
+var $listItem = document.createElement('li');
+$entriesList[0].appendChild($listItem);
```

Comment:
you should be rendering `li`s in the render function, you dont want to do something like this out in the non function space. you can eliminate this code

Reply:
Updated render function so that the `li` is created within the function (lines 51, 54 of main.js)